targets/ublk.nfs.cpp

```diff
@@ -50,9 +103,29 @@ static void nfs_handle_event(const struct ublksrv_queue *q)
 		cb_data = tmp;
 	}
 
-	pthread_spin_lock(&q_data->io_list_lock);
-	ublksrv_queue_handled_event(q);
-	pthread_spin_unlock(&q_data->io_list_lock);
+	if (!use_mshot_evt(q_data)) {
+		pthread_spin_lock(&q_data->io_list_lock);
+		ublksrv_queue_handled_event(q);
```

Comment:
This is from the old code,  I was just about to send a change to remove the spinlock as I don't think we need it.   Can you remove the spinlock in your PR?